developer-extension/src/panel/hooks/useEvents.ts

```diff
@@ -22,17 +22,26 @@ export function useEvents() {
     query: '',
   })
 
-  useEffect(
-    () =>
-      listenRequests((newEvents) => {
-        setEvents((oldEvents) =>
-          [...newEvents, ...oldEvents]
-            .sort((first: any, second: any) => second.date - first.date)
-            .slice(0, MAXIMUM_LOGGED_EVENTS)
-        )
-      }),
-    []
-  )
+  useEffect(() => {
+    const removeListener = listenRequests((newEvents) => {
+      setEvents((oldEvents) =>
+        [...newEvents, ...oldEvents]
+          .sort((first: any, second: any) => second.date - first.date)
+          .slice(0, MAXIMUM_LOGGED_EVENTS)
+      )
+    })
+
+    const clearCurrentEvents = (details: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+      if (['reload'].includes(details.transitionType)) setEvents([])
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
      if (details.transitionType === 'reload') setEvents([])
```